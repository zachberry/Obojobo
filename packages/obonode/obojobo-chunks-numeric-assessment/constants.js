const NUMERIC_ASSESSMENT_NODE = 'ObojoboDraft.Chunks.NumericAssessment'
const NUMERIC_ANSWER_NODE = 'ObojoboDraft.Chunks.NumericAssessment.NumericAsnwer'
const NUMERIC_CHOICE_NODE = 'ObojoboDraft.Chunks.NumericAssessment.NumericChoice'
const NUMERIC_FEEDBACK_NODE = 'ObojoboDraft.Chunks.NumericAssessment.NumericFeedback'

const EXACT_ANSWER = 'Exact answer'
const EXACT = 'exact'
const MARGIN_OF_ERROR = 'Margin of error'
const MARGIN = 'margin'
const WITHIN_A_RANGE = 'Within a range'
const RANGE = 'range'
const PRECISE_RESPONSE = 'Precise response'
const PRECISE = 'precise'
const PERCENT = 'Percent'
const PERCENT_S = 'percent'
const ABSOLUTE = 'Absolute'
const ABSOLUTE_S = 'absolute'
const SIGNIFICANT_DIGITS = 'Significant digits'
const SIG_FIGS = 'sig-figs'
const DECIMAL_PLACES = 'Decimal places'
const DECIMALS = 'decimals'

const fullTextToSimplifed = {
	[EXACT_ANSWER]: EXACT,
	[MARGIN_OF_ERROR]: MARGIN,
	[WITHIN_A_RANGE]: RANGE,
	[PRECISE_RESPONSE]: PRECISE,
	[PERCENT]: PERCENT_S,
	[ABSOLUTE]: ABSOLUTE_S,
	[SIGNIFICANT_DIGITS]: SIG_FIGS,
	[DECIMAL_PLACES]: DECIMALS
}

const simplifedToFullText = {
	[EXACT]: EXACT_ANSWER,
	[MARGIN]: MARGIN_OF_ERROR,
	[RANGE]: WITHIN_A_RANGE,
	[PRECISE]: PRECISE_RESPONSE,
	[PERCENT_S]: PERCENT,
	[ABSOLUTE_S]: ABSOLUTE,
	[SIG_FIGS]: SIGNIFICANT_DIGITS,
	[DECIMALS]: DECIMAL_PLACES
}

const requirementDropdown = [EXACT_ANSWER, MARGIN_OF_ERROR, WITHIN_A_RANGE, PRECISE_RESPONSE]
const marginDropdown = [PERCENT, ABSOLUTE]
const precisionDropdown = [SIGNIFICANT_DIGITS, DECIMAL_PLACES]

export {
	NUMERIC_ASSESSMENT_NODE,
	NUMERIC_CHOICE_NODE,
	NUMERIC_ANSWER_NODE,
	NUMERIC_FEEDBACK_NODE,
	EXACT_ANSWER,
	MARGIN_OF_ERROR,
	WITHIN_A_RANGE,
	PRECISE_RESPONSE,
	PERCENT,
	ABSOLUTE,
	SIGNIFICANT_DIGITS,
	DECIMAL_PLACES,
	fullTextToSimplifed,
	simplifedToFullText,
	requirementDropdown,
	marginDropdown,
	precisionDropdown
}
