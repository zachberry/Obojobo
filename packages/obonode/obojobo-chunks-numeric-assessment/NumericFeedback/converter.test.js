jest.mock('obojobo-document-engine/src/scripts/common/index', () => ({
	Registry: {
		getItemForType: () => ({
			slateToObo: jest.fn(),
			oboToSlate: jest.fn()
		})
	}
}))

import Converter from './converter'
import { NUMERIC_FEEDBACK_NODE } from '../constants'

describe('NumericFeedback Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => null
			},
			nodes: [
				{
					type: 'oboeditor.component',
					nodes: [
						{
							type: 'mockNode'
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			children: [
				{
					type: NUMERIC_FEEDBACK_NODE
				},
				{
					type: 'notADefinedNode'
				}
			]
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
