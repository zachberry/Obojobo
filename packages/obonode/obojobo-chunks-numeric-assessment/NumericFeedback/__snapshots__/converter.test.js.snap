// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericFeedback Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": undefined,
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "nodes": Array [
        undefined,
      ],
      "object": "block",
      "type": "oboeditor.component",
    },
    Object {
      "nodes": Array [
        undefined,
      ],
      "object": "block",
      "type": "oboeditor.component",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`NumericFeedback Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [
    undefined,
  ],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;
